### PR TITLE
fix: Update .d.ts file to match the actual code

### PR DESCRIPTION
**Pull request by MatrixFrog**
#### Description of changes

A couple of functions take an optional `clientMetadata` parameter which is not documented in this .d.ts. This PR adds that parameter.
 
#### Description of how you validated changes

Tests pass.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazon-cognito-identity-js/index.d.ts [hunk 1]
```diff
@@ -115,7 +115,8 @@ declare module 'amazon-cognito-identity-js' {
 		public changePassword(
 			oldPassword: string,
 			newPassword: string,
-			callback: NodeCallback<Error, 'SUCCESS'>
+			callback: NodeCallback<Error, 'SUCCESS'>,
+			clientMetadata: any
```
Comment (MatrixFrog) [inline packages/amazon-cognito-identity-js/index.d.ts:119]: I just put `any` here out of laziness, to be honest, but maybe someone can suggest a more specific type :)

Comment (evcodes) [inline packages/amazon-cognito-identity-js/index.d.ts:119]: This works:
```suggestion
			clientMetadata?: ClientMetadata
```

File: packages/amazon-cognito-identity-js/index.d.ts [hunk 2]
```diff
@@ -207,7 +209,8 @@ declare module 'amazon-cognito-identity-js' {
 		): void;
 		public updateAttributes(
 			attributes: (CognitoUserAttribute | ICognitoUserAttributeData)[],
-			callback: NodeCallback<Error, string>
+			callback: NodeCallback<Error, string>,
+			clientMetadata: any
```
Comment (evcodes) [inline packages/amazon-cognito-identity-js/index.d.ts:213]: Same thing here:
```suggestion
			clientMetadata?: ClientMetadata
```

File: packages/amazon-cognito-identity-js/index.d.ts [hunk 3]
```diff
@@ -219,7 +222,8 @@ declare module 'amazon-cognito-identity-js' {
 				onSuccess: (success: string) => void;
 				onFailure: (err: Error) => void;
 				inputVerificationCode?: (data: string) => void | null;
-			}
+			},
+			clientMetadata: any
```
Comment (evcodes) [inline packages/amazon-cognito-identity-js/index.d.ts:226]:
```suggestion
			clientMetadata?: ClientMetadata
```

**Review by evcodes — CHANGES_REQUESTED** [review]
I added some comments. There is another PR that addressed one of your methods but the others are not handled yet so we can merge this once you get your changes across the line

**Comment by MatrixFrog** [comment]
thanks, I think this should be good now :)

**Review by haverchuck — DISMISSED** [review]

**Review by AllanZhengYP — DISMISSED** [review]
Confirmed the d.ts change aligns with the JSDoc. Looks good to me!

**Review by haverchuck — APPROVED** [review]

**Review by AllanZhengYP — APPROVED** [review]

**Pull request merged**